### DISCUSSION TITLE
How about adding 65816 support i.e. Apple IIGS and SNES

### DISCUSSION THREAD

**Discussion by snakebyte69** [cc65/cc65#1527] [Ideas]
Given its close relation to the 6502 would this be hard to add?

**Comment by oliverschmidt** [comment]
Check out e.g. https://cc65.github.io/mailarchive/2002-04/1255.html

**Reply by Fabrizio-Caruso** [reply]
@oliverschmidt I think the problem is that it is a lot of work that someone has to do...
Maybe one could use some of the code some other MOS65816 cross-compilers use but this is a wild idea.

Nevertheless I think some of the statements made in 2002 are no longer valid, i.e.,
"...
there are no emulators for existing 65816 machines (maybe I'm wrong here).
This means that development is much more costly, because one cannot use all
the development tools available on the PC, the debugger that usually comes
with an emulator and so on.
..."

So, it could be done but it is a huge amount of work.

**Comment by nlovdahl** [comment]
Is there any precedent to guide how "a huge amount of work" would get cut up? I certainly can't imagine one monolithic merge throwing everything in all at once. A couple of years ago I wrote a bunch of assembly for code generation that shot off from codegen.c, but then I lost all momentum for external reasons. Still, I think it's not an insubstantial chunk of what needs to be done if I understand the compiler's setup correctly. I'm not sure how someone would want to structure a roadmap for this sort of work though.

**Comment by mrdudz** [comment]
Well, whoever wanted to tackle this would likely work in his own branch - and then indeed we'd merge it all at once when it is ready and tested. There no "official" plans for this (or anything else) for that matter

**Comment by nlovdahl** [comment]
Isn't that a bit much for one go?

**Reply by polluks** [reply]
How about a smooth transition?
1. stage: accu 16-bit
2. stage: indices 16-bit
3. stage: 24-bit addressing

**Comment by mrdudz** [comment]
Well, the problem here is that no one is actively developing cc65. And i wouldn't want to merge anything that doesn't even work. Of course if there are smaller logical steps that make sense to be merged separately, that can be done too
